src/lib/execPromise.ts

```diff
@@ -126,32 +140,39 @@ export const execPromise = ({
       }
 
       if (stderr.includes("linker `cc` not found")) {
-        rej(
-          new CLIError(
-            `\n${color.yellow(MARINE_CARGO_DEPENDENCY)} requires ${color.yellow(
-              "build-essential"
-            )} to be installed. Please install it and try again.\nOn debian-based systems (e.g. Ubuntu) you can install it using this command:\n\n${color.yellow(
-              "sudo apt install build-essential"
-            )}\n`
-          )
-        );
+        res({
+          expectedErrorMessage: `\n${color.yellow(
+            MARINE_CARGO_DEPENDENCY
+          )} requires ${color.yellow(
+            "build-essential"
+          )} to be installed. Please install it and try again.\nOn debian-based systems (e.g. Ubuntu) you can install it using this command:\n\n${color.yellow(
+            "sudo apt install build-essential"
+          )}\n`,
+        });
 
         return;
       }
 
       if (code !== 0) {
-        rej(
-          new Error(
-            `${printOutput ? "" : stderr}\nprocess exited${
-              code === null ? "" : ` with code ${code}`
-            }\n${debugInfo}`
-          )
-        );
-
-        return;
+        res({
+          errorMessage: `${getCommandFailedMessage(code)}${getErrorMessage(
+            printOutput,
+            stderr
+          )}`,
+        });
       }
 
-      res(stdout);
+      res({ success: stdout });
     });
   });
+
+  if ("expectedErrorMessage" in result) {
+    throw new CLIError(result.expectedErrorMessage);
+  }
+
+  if ("errorMessage" in result) {
+    throw new Error(result.errorMessage);
+  }
```

Comment:
What is `expectedErrorMessage` and why is it different from `errorMessage`?

Reply:
expected error message (CLIError) is the same thing that happens when you do commandObj.error - it's just an error that is expected to be encountered by the user of the CLI. When expected error happens CLI doesn't print stack trace and also it is sent to Countly in a different way

Reply:
so that's basically a non-fatal error? I've seen languages separate Fatal error from others.

maybe name fields as `fatalError` and `nonFatalError`? not sure

Reply:
they are all fatal, meaining that after they happen cli stops execution. The difference really is if the errors are expected or not. If they are expected then the error message is nice and user knows what to do next and we don't need to fix anything. But unexpected errors are something that should not normally happen

Reply:
okay. Let's at least document it in the code?

Reply:
I imporved the code - check it out and tell me if any additional documentation is required